[Index Patterns Field Formatter] Added human readable precise formatter for duration

shahzad31 opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/94641
Closes https://github.com/elastic/kibana/issues/11706

Added a new field formatter in index patterns for duration field to be used in observability use cases

Need to fix for https://github.com/elastic/uptime/issues/327

Also added a use short suffix check box to use short suffix strings in units.

Also aded an option whether to include space between suffix /value

This is how it works on sample data

![image](https://user-images.githubusercontent.com/3505601/120321797-02308b80-c2e4-11eb-9fc5-6b1954ce4335.png)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

formgeist commented on src/plugins/data/common/field_formats/converters/duration.ts:
```diff
@@ -100,50 +100,80 @@ const outputFormats = [
       defaultMessage: 'Seconds',
     }),
+    shortText: i18n.translate('data.fieldFormats.duration.outputFormats.asSeconds.short', {
+      defaultMessage: 'sec',
```
Come to think of it, I think we might be using `s` instead of `sec` (at least in APM). I think both are fine, I just prefer `s` because it's shorter and will fit better in more condensed usages, like tables and charts.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/)
* Commit: d006e993f5b73a321f86dd478c3aed3e97c43ce1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/d006e993f5b73a321f86dd478c3aed3e97c43ce1)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/818fa90fbbec3f75d9603becd25098896f45b091) | [after](https://github.com/elastic/kibana/commit/d006e993f5b73a321f86dd478c3aed3e97c43ce1) | diff |
| --- | --- | --- | --- |
| `data` | 3263 | 3264 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/818fa90fbbec3f75d9603becd25098896f45b091) | [after](https://github.com/elastic/kibana/commit/d006e993f5b73a321f86dd478c3aed3e97c43ce1) | diff |
| --- | --- | --- | --- |
| `data` | 817.4KB | 820.4KB | +3.0KB |
| `indexPatternFieldEditor` | 82.0KB | 82.6KB | +579.0B |
| total |  |  | +3.6KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/818fa90fbbec3f75d9603becd25098896f45b091) | [after](https://github.com/elastic/kibana/commit/d006e993f5b73a321f86dd478c3aed3e97c43ce1) | diff |
| --- | --- | --- | --- |
| `data` | 3763 | 3764 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/818fa90fbbec3f75d9603becd25098896f45b091) | [after](https://github.com/elastic/kibana/commit/d006e993f5b73a321f86dd478c3aed3e97c43ce1) | diff |
| --- | --- | --- | --- |
| `canvas` | 29 | 25 | -4 |
| `crossClusterReplication` | 8 | 6 | -2 |
| `fleet` | 22 | 20 | -2 |
| `globalSearch` | 4 | 2 | -2 |
| `indexManagement` | 12 | 7 | -5 |
| `infra` | 261 | 149 | -112 |
| `lens` | 67 | 45 | -22 |
| `licensing` | 18 | 15 | -3 |
| `lists` | 239 | 236 | -3 |
| `maps` | 286 | 208 | -78 |
| `ml` | 121 | 115 | -6 |
| `monitoring` | 109 | 56 | -53 |
| `securitySolution` | 386 | 342 | -44 |
| `stackAlerts` | 101 | 95 | -6 |
| total |  |  | -342 |


</details>

### History
* :broken_heart: [Build #127081](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/) failed 3ea640afa945644889f48c892efc48978a778afc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/","number":"127081","commit":"3ea640afa945644889f48c892efc48978a778afc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"}]}
PIPELINE-->

wylieconlon reviewed (changes requested):
Hi @shahzad31 thanks for taking this on! The concept is pretty much exactly what users are requesting, and here's my feedback after testing it:

* Values less than 1 are bugged in this formatter, so just to take one example: with `nanoseconds` input, `123` nanoseconds should either display as `123 ns`, `1.23 µs`, or `0.000123 ms`- but currently displays `0`. Same with negative values. I would expect that for values less than 1, the suffix would be chosen based on the number of decimal places that are displayed. So if I choose to display 6 decimal places I would see `0.000123 ms`, but if I choose 0 decimal places it should be `123 ns`.
* Suffix should be required for this new formatter, same as the human readable
* The name "dynamic" is not very clear. I suggest renaming "human readable" -> approximate, and "dynamic" -> precise
* I suggest moving this new format option to be near the top of the list, not the bottom
* The value 0 should get a suffix too, as `0 ms` is still a unit.
* Decimal places aren't affecting the value `0`, so if I have 2 decimals I want to see `0.00 ms`
* The suffixes you've chosen are inconsistent lengths and case, for example you have `d`, `sec`, and `Yr`

It would be great if the unit tests would test multiple cases per format setting, so that you can test the whole range of formats from negative, less than 1, very large, etc.

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServices)

shahzad31 commented:
> Hi @shahzad31 thanks for taking this on! The concept is pretty much exactly what users are requesting, and here's my feedback after testing it:
> 
> * Values less than 1 are bugged in this formatter, so just to take one example: with `nanoseconds` input, `123` nanoseconds should either display as `123 ns`, `1.23 µs`, or `0.000123 ms`- but currently displays `0`. Same with negative values. I would expect that for values less than 1, the suffix would be chosen based on the number of decimal places that are displayed. So if I choose to display 6 decimal places I would see `0.000123 ms`, but if I choose 0 decimal places it should be `123 ns`.

i have addressed this

> * Suffix should be required for this new formatter, same as the human readable

fixed as well

> * The value 0 should get a suffix too, as `0 ms` is still a unit.

Fixed

> * Decimal places aren't affecting the value `0`, so if I have 2 decimals I want to see `0.00 ms`

Fixed

> * The suffixes you've chosen are inconsistent lengths and case, for example you have `d`, `sec`, and `Yr`

I have tried improving this

> It would be great if the unit tests would test multiple cases per format setting, so that you can test the whole range of formats from negative, less than 1, very large, etc.

added few more tests

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/)
* Commit: 848fd016320f864a3997cca2d95520a1164568e1
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/execution/node/425/log)

### History
* :green_heart: [Build #127091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/) succeeded d006e993f5b73a321f86dd478c3aed3e97c43ce1
* :broken_heart: [Build #127081](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/) failed 3ea640afa945644889f48c892efc48978a778afc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/","number":"127081","commit":"3ea640afa945644889f48c892efc48978a778afc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/)
* Commit: cec65215ce366eba30c16775f9777aecdc34e240
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/cec65215ce366eba30c16775f9777aecdc34e240)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/31778b31c272e0e184b53480515dee5f6d42c94a) | [after](https://github.com/elastic/kibana/commit/cec65215ce366eba30c16775f9777aecdc34e240) | diff |
| --- | --- | --- | --- |
| `data` | 3263 | 3264 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/31778b31c272e0e184b53480515dee5f6d42c94a) | [after](https://github.com/elastic/kibana/commit/cec65215ce366eba30c16775f9777aecdc34e240) | diff |
| --- | --- | --- | --- |
| `data` | 817.4KB | 820.7KB | +3.3KB |
| `indexPatternFieldEditor` | 82.0KB | 83.3KB | +1.3KB |
| total |  |  | +4.6KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/31778b31c272e0e184b53480515dee5f6d42c94a) | [after](https://github.com/elastic/kibana/commit/cec65215ce366eba30c16775f9777aecdc34e240) | diff |
| --- | --- | --- | --- |
| `data` | 3763 | 3764 | +1 |


</details>

### History
* :broken_heart: [Build #127434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/) failed 848fd016320f864a3997cca2d95520a1164568e1
* :green_heart: [Build #127091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/) succeeded d006e993f5b73a321f86dd478c3aed3e97c43ce1
* :broken_heart: [Build #127081](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/) failed 3ea640afa945644889f48c892efc48978a778afc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/","number":"127081","commit":"3ea640afa945644889f48c892efc48978a778afc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/)
* Commit: 55d05111906e2a5cbde5a01f4824c447a260a455
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/55d05111906e2a5cbde5a01f4824c447a260a455)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/c5aa39835de16357553b4fe1b925a55be4999cc9) | [after](https://github.com/elastic/kibana/commit/55d05111906e2a5cbde5a01f4824c447a260a455) | diff |
| --- | --- | --- | --- |
| `data` | 3264 | 3265 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c5aa39835de16357553b4fe1b925a55be4999cc9) | [after](https://github.com/elastic/kibana/commit/55d05111906e2a5cbde5a01f4824c447a260a455) | diff |
| --- | --- | --- | --- |
| `data` | 817.5KB | 820.9KB | +3.4KB |
| `indexPatternFieldEditor` | 81.8KB | 83.1KB | +1.3KB |
| total |  |  | +4.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/c5aa39835de16357553b4fe1b925a55be4999cc9) | [after](https://github.com/elastic/kibana/commit/55d05111906e2a5cbde5a01f4824c447a260a455) | diff |
| --- | --- | --- | --- |
| `data` | 3764 | 3765 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/c5aa39835de16357553b4fe1b925a55be4999cc9) | [after](https://github.com/elastic/kibana/commit/55d05111906e2a5cbde5a01f4824c447a260a455) | diff |
| --- | --- | --- | --- |
| `canvas` | 29 | 25 | -4 |
| `crossClusterReplication` | 8 | 6 | -2 |
| `fleet` | 22 | 20 | -2 |
| `globalSearch` | 4 | 2 | -2 |
| `indexManagement` | 12 | 7 | -5 |
| `infra` | 261 | 149 | -112 |
| `lens` | 67 | 45 | -22 |
| `licensing` | 18 | 15 | -3 |
| `lists` | 239 | 236 | -3 |
| `maps` | 286 | 208 | -78 |
| `ml` | 121 | 115 | -6 |
| `monitoring` | 109 | 56 | -53 |
| `securitySolution` | 386 | 342 | -44 |
| `stackAlerts` | 101 | 95 | -6 |
| total |  |  | -342 |


</details>

### History
* :green_heart: [Build #127395](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/) succeeded cec65215ce366eba30c16775f9777aecdc34e240
* :broken_heart: [Build #127434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/) failed 848fd016320f864a3997cca2d95520a1164568e1
* :green_heart: [Build #127091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/) succeeded d006e993f5b73a321f86dd478c3aed3e97c43ce1
* :broken_heart: [Build #127081](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/) failed 3ea640afa945644889f48c892efc48978a778afc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/","number":"127081","commit":"3ea640afa945644889f48c892efc48978a778afc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/","number":"127444","commit":"55d05111906e2a5cbde5a01f4824c447a260a455"}]}
PIPELINE-->

wylieconlon reviewed (commented):
@shahzad31 It looks like you've introduced a new bug in the UI, which is that the form options for the precise format are disabled. Also, I would have expected that the suffix option is not shown when using the precise format.

<img width="435" alt="Screen Shot 2021-05-26 at 7 54 53 PM" src="https://user-images.githubusercontent.com/666475/119745543-59ee7300-be5c-11eb-8a50-f78be48f0e29.png">

shahzad31 commented:
> @shahzad31 It looks like you've introduced a new bug in the UI, which is that the form options for the precise format are disabled. Also, I would have expected that the suffix option is not shown when using the precise format.
> 

@wylieconlon i fixed the bug, in case of human readable precise, show suffix isn't shown. Only the options to use short suffix is shown. Since that's still applicable.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/)
* Commit: 1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/77452e686be6a3b202e3a9e7bcf0fe0174a07ee9) | [after](https://github.com/elastic/kibana/commit/1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6) | diff |
| --- | --- | --- | --- |
| `data` | 3264 | 3265 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/77452e686be6a3b202e3a9e7bcf0fe0174a07ee9) | [after](https://github.com/elastic/kibana/commit/1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6) | diff |
| --- | --- | --- | --- |
| `data` | 817.5KB | 820.9KB | +3.4KB |
| `indexPatternFieldEditor` | 81.8KB | 83.1KB | +1.3KB |
| total |  |  | +4.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/77452e686be6a3b202e3a9e7bcf0fe0174a07ee9) | [after](https://github.com/elastic/kibana/commit/1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6) | diff |
| --- | --- | --- | --- |
| `data` | 3764 | 3765 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/77452e686be6a3b202e3a9e7bcf0fe0174a07ee9) | [after](https://github.com/elastic/kibana/commit/1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6) | diff |
| --- | --- | --- | --- |
| `ml` | 121 | 115 | -6 |
| `securitySolution` | 386 | 342 | -44 |
| total |  |  | -50 |


</details>

### History
* :green_heart: [Build #127444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/) succeeded 55d05111906e2a5cbde5a01f4824c447a260a455
* :green_heart: [Build #127395](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/) succeeded cec65215ce366eba30c16775f9777aecdc34e240
* :broken_heart: [Build #127434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/) failed 848fd016320f864a3997cca2d95520a1164568e1
* :green_heart: [Build #127091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/) succeeded d006e993f5b73a321f86dd478c3aed3e97c43ce1
* :broken_heart: [Build #127081](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/) failed 3ea640afa945644889f48c892efc48978a778afc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127081/","number":"127081","commit":"3ea640afa945644889f48c892efc48978a778afc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/","number":"127444","commit":"55d05111906e2a5cbde5a01f4824c447a260a455"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/","number":"127708","commit":"1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6"}]}
PIPELINE-->

wylieconlon reviewed (approved):
LGTM! This is great, users will be happy to see this.

mattkime reviewed (approved):
changes lgtm and work well

shahzad31 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/)
* Commit: dbfadda438c4985fb8a53a2eb721f83b3dd9efe8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/dbfadda438c4985fb8a53a2eb721f83b3dd9efe8)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/3ef24aa56c749723b8b2f99bfc1e71def123b57b) | [after](https://github.com/elastic/kibana/commit/dbfadda438c4985fb8a53a2eb721f83b3dd9efe8) | diff |
| --- | --- | --- | --- |
| `data` | 3223 | 3224 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/3ef24aa56c749723b8b2f99bfc1e71def123b57b) | [after](https://github.com/elastic/kibana/commit/dbfadda438c4985fb8a53a2eb721f83b3dd9efe8) | diff |
| --- | --- | --- | --- |
| `data` | 817.7KB | 821.1KB | +3.4KB |
| `indexPatternFieldEditor` | 81.8KB | 83.1KB | +1.3KB |
| total |  |  | +4.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/3ef24aa56c749723b8b2f99bfc1e71def123b57b) | [after](https://github.com/elastic/kibana/commit/dbfadda438c4985fb8a53a2eb721f83b3dd9efe8) | diff |
| --- | --- | --- | --- |
| `data` | 3768 | 3769 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/3ef24aa56c749723b8b2f99bfc1e71def123b57b) | [after](https://github.com/elastic/kibana/commit/dbfadda438c4985fb8a53a2eb721f83b3dd9efe8) | diff |
| --- | --- | --- | --- |
| `canvas` | 29 | 25 | -4 |
| `crossClusterReplication` | 8 | 6 | -2 |
| `fleet` | 22 | 20 | -2 |
| `globalSearch` | 4 | 2 | -2 |
| `indexManagement` | 12 | 7 | -5 |
| `infra` | 261 | 149 | -112 |
| `lens` | 67 | 45 | -22 |
| `licensing` | 18 | 15 | -3 |
| `lists` | 239 | 236 | -3 |
| `maps` | 286 | 208 | -78 |
| `ml` | 121 | 115 | -6 |
| `monitoring` | 109 | 56 | -53 |
| `securitySolution` | 390 | 346 | -44 |
| `stackAlerts` | 101 | 95 | -6 |
| total |  |  | -342 |


</details>

### History
* :green_heart: [Build #127708](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/) succeeded 1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6
* :green_heart: [Build #127444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/) succeeded 55d05111906e2a5cbde5a01f4824c447a260a455
* :green_heart: [Build #127395](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/) succeeded cec65215ce366eba30c16775f9777aecdc34e240
* :broken_heart: [Build #127434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/) failed 848fd016320f864a3997cca2d95520a1164568e1
* :green_heart: [Build #127091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/) succeeded d006e993f5b73a321f86dd478c3aed3e97c43ce1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127091/","number":"127091","commit":"d006e993f5b73a321f86dd478c3aed3e97c43ce1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/","number":"127444","commit":"55d05111906e2a5cbde5a01f4824c447a260a455"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/","number":"127708","commit":"1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/","number":"128131","commit":"dbfadda438c4985fb8a53a2eb721f83b3dd9efe8"}]}
PIPELINE-->

Dosant commented on src/plugins/data/common/field_formats/converters/duration.ts:
```diff
@@ -89,59 +90,89 @@ const inputFormats = [
-  text: i18n.translate('data.fieldFormats.duration.outputFormats.humanize', {
-    defaultMessage: 'Human Readable',
+  text: i18n.translate('data.fieldFormats.duration.outputFormats.humanize.precise', {
+    defaultMessage: 'Human Readable precise',
```
Nit copy: I think casing should be consistent here. It should either be `Human Readable Precise` or `Human readable precise`. 

As I typing it out autocorrect actually suggesting me that this needs a hyphen. Then how about: 

`Human-readable (precise)` and `Human-readable (approximate)` ?

shahzad31 replied:
I liked the suggestions. Done

Dosant reviewed (commented):
1. If I understood correctly then the old `Human readable` should become `Human-readable approximate`? But it looks like that when I check out this pr old fields that were `Human readable` now use `Human-readable precise`, hence their format in apps change. 
Is this behavior expected? I'd assume old fields should follow old formatting unless changed explicitly by a user? 


2. I noticed some inconsistencies between formatting when precise vs approximate are used: 
For example, see `hours` vs `Hours`? Would it be difficult to improve? 

![Screen Shot 2021-05-31 at 14 25 52](https://user-images.githubusercontent.com/7784120/120195071-a3eda500-c21e-11eb-9816-3fb32d33aece.png)
![Screen Shot 2021-05-31 at 14 41 40](https://user-images.githubusercontent.com/7784120/120195079-a5b76880-c21e-11eb-8f82-3c57c8a08ff2.png)


3. Also per the images above, looks like `precise` returns an empty string when duration is `0` where `approximate` (old behavior) returns `a few seconds`. I would agree that the old behavior is incorrect, but I also probably empty string should return something different? Or is this expected?

shahzad31 commented:
@Dosant so weirdly in case of default output format it doesn't set the output format value in the saved object, it goes as empty. This is why the behavior you are seeing.  Will have to find some solution for this. Or otherwise i will leave the default output format as Human readable approx. 

> 1. If I understood correctly then the old `Human readable` should become `Human-readable approximate`? But it looks like that when I check out this pr old fields that were `Human readable` now use `Human-readable precise`, hence their format in apps change.
>    Is this behavior expected? I'd assume old fields should follow old formatting unless changed explicitly by a user?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128248/)
* Commit: a7a0d49c4b22a045c5609cff425cd378c782d4e4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/a7a0d49c4b22a045c5609cff425cd378c782d4e4)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 3223 | 3224 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 817.4KB | 820.8KB | +3.4KB |
| `indexPatternFieldEditor` | 81.8KB | 83.1KB | +1.3KB |
| total |  |  | +4.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 3768 | 3769 | +1 |


</details>

### History
* :green_heart: [Build #128131](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/) succeeded dbfadda438c4985fb8a53a2eb721f83b3dd9efe8
* :green_heart: [Build #127708](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/) succeeded 1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6
* :green_heart: [Build #127444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/) succeeded 55d05111906e2a5cbde5a01f4824c447a260a455
* :green_heart: [Build #127395](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/) succeeded cec65215ce366eba30c16775f9777aecdc34e240
* :broken_heart: [Build #127434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/) failed 848fd016320f864a3997cca2d95520a1164568e1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127434/","number":"127434","commit":"848fd016320f864a3997cca2d95520a1164568e1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/","number":"127444","commit":"55d05111906e2a5cbde5a01f4824c447a260a455"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/","number":"127708","commit":"1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/","number":"128131","commit":"dbfadda438c4985fb8a53a2eb721f83b3dd9efe8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128248/","number":"128248","commit":"a7a0d49c4b22a045c5609cff425cd378c782d4e4"}]}
PIPELINE-->

shahzad31 commented:
@Dosant i have reverted the default format to Human-readable approximate to avoid complexities. We can tackle that in a follow up if needs be.

Dosant reviewed (approved):
Retested recent changes. LGTM 👍

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/101046)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/101046">](https://github.com/elastic/kibana/pull/101046) |

This backport PR will be merged automatically after passing CI.

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128315/)
* Commit: a7a0d49c4b22a045c5609cff425cd378c782d4e4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-100540/a7a0d49c4b22a045c5609cff425cd378c782d4e4)

* Flaky suites:
  * xpack-kibana-ciGroup6

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128315/testReport)

<details><summary>Kibana Pipeline / general / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/index_management/indices·js.apis management index management indices reload (not on Cloud) should list all the indices with the expected properties and data enrichers</summary>

[Link to Jenkins](https://kibana-ci.elastic.co//job/elastic+kibana+pipeline-pull-request/128315/testReport/junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_index_management_indices·js/Kibana_Pipeline___general___apis_management_index_management_indices_reload__not_on_Cloud__should_list_all_the_indices_with_the_expected_properties_and_data_enrichers)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 6 times on tracked branches: https://github.com/elastic/kibana/issues/90565

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook in "apis"
[00:05:45]           └-: management
[00:05:45]             └-> "before all" hook in "management"
[00:06:03]             └-: index management
[00:06:03]               └-> "before all" hook in "index management"
[00:06:03]               └-: indices
[00:06:03]                 └-> "before all" hook in "indices"
[00:06:05]                 └-: reload
[00:06:05]                   └-> "before all" hook for "should allow reloading only certain indices"
[00:06:05]                   └-> should allow reloading only certain indices
[00:06:05]                     └-> "before each" hook: global before each for "should allow reloading only certain indices"
[00:06:05]                     │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-18-tests-xxl-1622549862751015791] [gsegnlzmcvmmjdwatl-1622552779684] creating index, cause [api], templates [], shards [1]/[1]
[00:06:05]                     └- ✓ pass  (67ms) "apis management index management indices reload should allow reloading only certain indices"
[00:06:05]                   └-: (not on Cloud)
[00:06:05]                     └-> "before all" hook for "should list all the indices with the expected properties and data enrichers"
[00:06:05]                     └-> should list all the indices with the expected properties and data enrichers
[00:06:05]                       └-> "before each" hook: global before each for "should list all the indices with the expected properties and data enrichers"
[00:06:05]                       └- ✖ fail: apis management index management indices reload (not on Cloud) should list all the indices with the expected properties and data enrichers
[00:06:05]                       │       Error: expected [ 'aliases',
[00:06:05]                       │   'data_stream',
[00:06:05]                       │   'documents',
[00:06:05]                       │   'health',
[00:06:05]                       │   'hidden',
[00:06:05]                       │   'ilm',
[00:06:05]                       │   'isFollowerIndex',
[00:06:05]                       │   'isFrozen',
[00:06:05]                       │   'isRollupIndex',
[00:06:05]                       │   'name',
[00:06:05]                       │   'primary',
[00:06:05]                       │   'replica',
[00:06:05]                       │   'size',
[00:06:05]                       │   'status',
[00:06:05]                       │   'uuid' ] to sort of equal [ 'aliases',
[00:06:05]                       │   'documents',
[00:06:05]                       │   'health',
[00:06:05]                       │   'hidden',
[00:06:05]                       │   'ilm',
[00:06:05]                       │   'isFollowerIndex',
[00:06:05]                       │   'isFrozen',
[00:06:05]                       │   'isRollupIndex',
[00:06:05]                       │   'name',
[00:06:05]                       │   'primary',
[00:06:05]                       │   'replica',
[00:06:05]                       │   'size',
[00:06:05]                       │   'status',
[00:06:05]                       │   'uuid' ]
[00:06:05]                       │       + expected - actual
[00:06:05]                       │ 
[00:06:05]                       │        [
[00:06:05]                       │          "aliases"
[00:06:05]                       │       -  "data_stream"
[00:06:05]                       │          "documents"
[00:06:05]                       │          "health"
[00:06:05]                       │          "hidden"
[00:06:05]                       │          "ilm"
[00:06:05]                       │       
[00:06:05]                       │       at Assertion.assert (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/expect/expect.js:100:11)
[00:06:05]                       │       at Assertion.eql (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/expect/expect.js:244:8)
[00:06:05]                       │       at Context.<anonymous> (test/api_integration/apis/management/index_management/indices.js:237:41)
[00:06:05]                       │       at Object.apply (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:16)
[00:06:05]                       │ 
[00:06:05]                       │ 

```

#### Stack Trace
```
Error: expected [ 'aliases',
  'data_stream',
  'documents',
  'health',
  'hidden',
  'ilm',
  'isFollowerIndex',
  'isFrozen',
  'isRollupIndex',
  'name',
  'primary',
  'replica',
  'size',
  'status',
  'uuid' ] to sort of equal [ 'aliases',
  'documents',
  'health',
  'hidden',
  'ilm',
  'isFollowerIndex',
  'isFrozen',
  'isRollupIndex',
  'name',
  'primary',
  'replica',
  'size',
  'status',
  'uuid' ]
    at Assertion.assert (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/expect/expect.js:100:11)
    at Assertion.eql (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/expect/expect.js:244:8)
    at Context.<anonymous> (test/api_integration/apis/management/index_management/indices.js:237:41)
    at Object.apply (/dev/shm/workspace/parallel/8/kibana/node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:16) {
  actual: '[\n' +
    '  "aliases"\n' +
    '  "data_stream"\n' +
    '  "documents"\n' +
    '  "health"\n' +
    '  "hidden"\n' +
    '  "ilm"\n' +
    '  "isFollowerIndex"\n' +
    '  "isFrozen"\n' +
    '  "isRollupIndex"\n' +
    '  "name"\n' +
    '  "primary"\n' +
    '  "replica"\n' +
    '  "size"\n' +
    '  "status"\n' +
    '  "uuid"\n' +
    ']',
  expected: '[\n' +
    '  "aliases"\n' +
    '  "documents"\n' +
    '  "health"\n' +
    '  "hidden"\n' +
    '  "ilm"\n' +
    '  "isFollowerIndex"\n' +
    '  "isFrozen"\n' +
    '  "isRollupIndex"\n' +
    '  "name"\n' +
    '  "primary"\n' +
    '  "replica"\n' +
    '  "size"\n' +
    '  "status"\n' +
    '  "uuid"\n' +
    ']',
  showDiff: true
}
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 3223 | 3224 | +1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 817.4KB | 820.8KB | +3.4KB |
| `indexPatternFieldEditor` | 81.8KB | 83.1KB | +1.3KB |
| total |  |  | +4.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/af4b8e66263d3758ed8a3acb9c761e0e12fd5e60) | [after](https://github.com/elastic/kibana/commit/a7a0d49c4b22a045c5609cff425cd378c782d4e4) | diff |
| --- | --- | --- | --- |
| `data` | 3768 | 3769 | +1 |


</details>

### History
* :green_heart: [Build #128248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128248/) succeeded a7a0d49c4b22a045c5609cff425cd378c782d4e4
* :green_heart: [Build #128131](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/) succeeded dbfadda438c4985fb8a53a2eb721f83b3dd9efe8
* :green_heart: [Build #127708](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/) succeeded 1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6
* :green_heart: [Build #127444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/) succeeded 55d05111906e2a5cbde5a01f4824c447a260a455
* :green_heart: [Build #127395](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/) succeeded cec65215ce366eba30c16775f9777aecdc34e240

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127395/","number":"127395","commit":"cec65215ce366eba30c16775f9777aecdc34e240"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127444/","number":"127444","commit":"55d05111906e2a5cbde5a01f4824c447a260a455"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/127708/","number":"127708","commit":"1cef50e99a2804e2f55d1b3f945dfbd8d34d8ee6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128131/","number":"128131","commit":"dbfadda438c4985fb8a53a2eb721f83b3dd9efe8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128248/","number":"128248","commit":"a7a0d49c4b22a045c5609cff425cd378c782d4e4"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/128315/","number":"128315","commit":"a7a0d49c4b22a045c5609cff425cd378c782d4e4"}]}
PIPELINE-->
